libcontainer/container_linux.go

```diff
@@ -112,28 +112,33 @@ func (c *Container) OCIState() (*specs.State, error) {
 	return c.currentOCIState()
 }
 
+// ignoreCgroupError filters out cgroup-related errors that can be ignored,
+// because the container is stopped and its cgroup is gone.
+func (c *Container) ignoreCgroupError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if c.runType() == Stopped && !c.cgroupManager.Exists() {
+		return nil
+	}
+	return err
+}
+
 // Processes returns the PIDs inside this container. The PIDs are in the
 // namespace of the calling process.
 //
 // Some of the returned PIDs may no longer refer to processes in the container,
 // unless the container state is PAUSED in which case every PID in the slice is
 // valid.
 func (c *Container) Processes() ([]int, error) {
-	var pids []int
-	status, err := c.currentStatus()
-	if err != nil {
-		return pids, err
-	}
-	// for systemd cgroup, the unit's cgroup path will be auto removed if container's all processes exited
-	if status == Stopped && !c.cgroupManager.Exists() {
+	pids, err := c.cgroupManager.GetAllPids()
+	if c.ignoreCgroupError(err) == nil {
```

Comment:
Maybe there is a more readable implement like this:
1. change the return type of `ignoreCgroupError` to `bool`;
2. if err == nil || c.ignoreCgroupError(err) {
Otherwise the reader needs to see the function body of `ignoreCgroupError` to find out what happened here.

Reply:
Oh well, my initial implementation was called `isIgnorableCgroupError` and returned `bool`. Then I took the approach used by `func ignoreTerminateErrors`.

Either way is fine with me.